pkg/agent/flowexporter/connections/l7_listener.go

```diff
@@ -114,22 +115,30 @@ func (l *L7Listener) listenAndAcceptConn() {
 	}
 	defer listener.Close()
 	klog.InfoS("L7 Listener Server started. Listening for connections...")
-	for {
-		conn, err := listener.Accept()
-		if err != nil {
-			klog.ErrorS(err, "Error accepting Suricata connection")
-			return
+	go func() {
+		for {
+			conn, err := listener.Accept()
+			if err != nil {
+				klog.ErrorS(err, "Error accepting Suricata connection")
+				return
```

Comment:
Previously, if `listener.Accept()` fails, the whole function will return and retry. With the new code, `listenAndAcceptConn` would just wait on stopCh and wouldn't retry.

Reply:
Thanks for catching this. I rewrote the function, PTAL.